### ISSUE TITLE
Favorite filter breaks Artists tab.

### ISSUE THREAD

**Issue by Liarra** [UnicornsOnLSD/finamp#72]
How to reproduce:
1. Go to Artists tab;
2. Press the star.

![image](https://user-images.githubusercontent.com/996691/125528340-2f91dce6-1f97-455f-a416-fd24d675c5d8.png)

Last log item:
```
[Flutter/SEVERE] 2021-07-13 23:29:07.262909: NoSuchMethodError: The method 'call' was called on null.
Receiver: null
Tried calling: call(Instance of 'Response')

#0 Object.noSuchMethod (dart:core-patch/object_patch.dart:54)
#1 _objectNoSuchMethod (dart:core-patch/object_patch.dart:97)
#2 _MusicScreenTabViewState._getPage (package:finamp/components/MusicScreen/MusicScreenTabView.dart:107)
```

**Comment by UnicornsOnLSD** [comment]
I'm aware of this, for some reason the request fails on Jellyfin's side (in the Jellyfin logs, you'll see an error). I need to look more into how Jellyfin Web gets artist favourites.

**Comment by wmyott** [comment]
Randomly looked into this a bit - testing the Jellyfin API via Postman it looks like the Artists/AlbumArtists API requires the userId query parameter for the Filter=IsFavorite option to work.

**Works:** GET https://{{JELLYFIN_URL}}/Artists/AlbumArtists?userId={{user_id}}&Filters=IsFavorite&api_key={{API_KEY}}

**Doesn't work:** GET https://{{JELLYFIN_URL}}/Artists/AlbumArtists?Filters=IsFavorite&api_key={{API_KEY}}

If the userId parameter is not there it returns a 500 and gives an error in the Jellyfin log (see below). My guess is that the Albums API does not require this because it has userId in the path already.

I'm not familiar enough with dart to try for a pull request myself, but hopefully this helps.

<details>
<summary>

```
[2022-03-11 14:19:43.375 -08:00] [ERR] [95] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL "GET" "/Artists/AlbumArtists".
Error: SQLitePCL.pretty.SQLiteException: no such column: IsFavorite
```

</summary>

```
   at SQLitePCL.pretty.SQLiteException.Throw(Int32 rc, Int32 extended, String msg)
   at SQLitePCL.pretty.SQLiteException.CheckOk(sqlite3 db, Int32 rc)
   at SQLitePCL.pretty.SQLiteDatabaseConnection.PrepareStatement(String sql, String& tail)
   at SQLitePCL.pretty.DatabaseConnection.PrepareStatement(IDatabaseConnection This, String sql)
   at Emby.Server.Implementations.Data.BaseSqliteRepository.PrepareAll(IDatabaseConnection connection, IReadOnlyList`1 sql)
   at Emby.Server.Implementations.Data.SqliteItemRepository.<>c__DisplayClass116_0.<GetItemValues>b__1(IDatabaseConnection db)
   at SQLitePCL.pretty.DatabaseConnection.<>c__DisplayClass20_0.<RunInTransaction>b__0(IDatabaseConnection db)
   at SQLitePCL.pretty.DatabaseConnection.RunInTransaction[T](IDatabaseConnection This, Func`2 f, TransactionMode mode)
   at SQLitePCL.pretty.DatabaseConnection.RunInTransaction(IDatabaseConnection This, Action`1 action, TransactionMode mode)
   at Emby.Server.Implementations.Data.SqliteItemRepository.GetItemValues(InternalItemsQuery query, Int32[] itemValueTypes, String returnType)
   at Emby.Server.Implementations.Data.SqliteItemRepository.GetAlbumArtists(InternalItemsQuery query)
   at Emby.Server.Implementations.Library.LibraryManager.GetAlbumArtists(InternalItemsQuery query)
   at Jellyfin.Api.Controllers.ArtistsController.GetAlbumArtists(Nullable`1 minCommunityRating, Nullable`1 startIndex, Nullable`1 limit, String searchTerm, Nullable`1 parentId, ItemFields[] fields, String[] excludeItemTypes, String[] includeItemTypes, ItemFilter[] filters, Nullable`1 isFavorite, String[] mediaTypes, String[] genres, Guid[] genreIds, String[] officialRatings, String[] tags, Int32[] years, Nullable`1 enableUserData, Nullable`1 imageTypeLimit, ImageType[] enableImageTypes, String person, Guid[] personIds, String[] personTypes, String[] studios, Guid[] studioIds, Nullable`1 userId, String nameStartsWithOrGreater, String nameStartsWith, String nameLessThan, Nullable`1 enableImages, Boolean enableTotalRecordCount)
   at lambda_method338(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

</details>

**Comment by UnicornsOnLSD** [comment]
I can confirm that this works! Fixed in 054034e.

Maybe this should be bought up with the Jellyfin devs? UserId is nullable in the documentation (https://api.jellyfin.org/#operation/GetAlbumArtists), so maybe there should at least be an assert for "if filters includes isfavourite, ensure that userid is not null" so that a useful error message is created.
